src/scenario-guide/azext_scenario_guide/custom.py

```diff
@@ -0,0 +1,360 @@
+
+    scope = SearchScope.get(scope)
+    match_rule = MatchRule.get(match_rule)
+    # Replacing "-" is to solve the problem that the search engine cannot match "-"
+    # e.g. `az scenario guide web-app create` => "web app create"
+    search_keyword = " ".join(map(lambda w: w.replace("-", " "), search_keyword))
+    results = search_online(search_keyword, scope, match_rule, top)
+
+    if not results:
+        send_feedback(scope, FeedbackOption.NO_RESULT, search_keyword)
+        print("\nSorry, no relevant E2E scenario examples found")
+        return
+
+    _show_search_item(results)
+
+    option_msg = [(Style.ACTION, " ? "), (Style.PRIMARY, "Please select your option "),
+                  (Style.SECONDARY, "(if none, enter 0)"), (Style.PRIMARY, ": ")]
+    option = select_option(option_msg, min_option=0, max_option=len(results), default_option=-1)
+    print()
+
+    if option == 0:
+        send_feedback(scope, FeedbackOption.NO_SELECT, search_keyword, results)
+        print('Thank you for your feedback. If you have more feedback, please submit it by using "az feedback" \n')
+        return
+
+    chosen_scenario = results[option - 1]
+    _show_detail(cmd, chosen_scenario)
+    send_feedback(scope, FeedbackOption.SELECT(option), search_keyword, results, chosen_scenario)
+
+    if cmd.cli_ctx.config.getboolean('scenario_guide', 'execute_in_prompt', fallback=True):
+        _execute_scenario(cmd, chosen_scenario)
+    else:
+        print('\nThank you for your feedback. AZ SCENARIO GUIDE is completed. '
+              'If you want to execute the commands in interactive mode, '
+              'you can use "az config set scenario_guide.execute_in_prompt=True" to set it up.\n')
+
+    return
+
+
+def _show_search_item(results):
+    """
+    Display searched scenarios in following format.
+
+    e.g.
+    [1] Monitor an App Service app with web server logs (4 Commands)
+    Include command: az webapp create
+    """
+
+    # To learn the structure of result,
+    # visit https://github.com/hackathon-cli-recommendation/cli-recommendation/blob/master/Docs/API_design_doc.md
+    for idx, result in enumerate(results):
+
+        print()
+        # display idx as "[ 1]" if max index is larger than 9
+        idx_str = f"[{idx+1:2}] " if len(results) >= 10 else f"[{idx+1:1}] "
+        num_notice = f" ({len(result['commandSet'])} Commands)"
+        print_styled_text([(Style.ACTION, idx_str), (Style.PRIMARY, result['scenario']), (Style.SECONDARY, num_notice)])
+
+        # Display a description or related command next line
+        # We will display the description if a matched keyword in description
+        # If there's no matched keyword in description, we will display the command with matched keyword instead
+        # If there's no matched keyword in commands, we will display the scenario name with matched keyword
+        # If there's no matched keyword in scenario name, we will display scenario description anyway
```

Comment:
```suggestion
        # Display the command description or related command in the secondary information
        # The specific display strategy are as follows:
        # 1. It will display the description if a matched keyword in description
        # 2. If there's no matched keyword in description, it will display the command with matched keyword instead
        # 3. If there's no matched keyword in commands, it will display the scenario name with matched keyword
        # 4. If there's no matched keyword in scenario name, it will display scenario description anyway
```